netsim-ffi/src/lib.rs

```diff
@@ -170,7 +179,7 @@ pub unsafe extern "C" fn netsim_socket_recv(
         // this is usually to signal it is time to release
         // the socket
         let _ = Box::from_raw(socket);
```

Comment:
don't `free` the socket though. I appreciate the comment above might have been the cause of this. But this is not the place to free the object. we have the function `netsim_socket_release` for that.